system-test/system_test_redis_integ.sh

```
@@ -0,0 +1,1264 @@
+
+# Run main function with timeout in CI mode
+# Only execute main if this script is being run directly (not sourced)
+if [ "${BASH_SOURCE[0]}" = "${0}" ]; then
+    if [ "$CI_MODE" = true ]; then
+        if command -v timeout >/dev/null 2>&1; then
+            # Use timeout but handle exit codes properly
+            if timeout "$TIMEOUT" bash -c 'source "$1"; main; exit $?' _ "$0"; then
+                # Test completed successfully within timeout
+                exit 0
+            else
+                local exit_code=$?
```

Shellcheck (0.11.0)

[error] 'local' is only valid in functions.

(SC2168)